packages/apollo-server-cache-redis/src/RedisClusterCache.ts

```diff
@@ -0,0 +1,58 @@
+import { KeyValueCache } from 'apollo-server-caching';
+import Redis, {
+  ClusterOptions,
+  ClusterNode,
+  Redis as RedisInstance,
+} from 'ioredis';
+import DataLoader from 'dataloader';
+
+export class RedisClusterCache implements KeyValueCache {
+  readonly client: any;
+  readonly defaultSetOptions = {
+    ttl: 300,
+  };
+
+  private loader: DataLoader<string, string>;
+
+  constructor(nodes: ClusterNode[], options?: ClusterOptions) {
+    this.client = new Redis.Cluster(nodes, options);
+
+    this.loader = new DataLoader(
+      (keys = []) =>
+        Promise.all(keys.map(key => this.client.get(key).catch(() => null))),
+      { cache: false },
+    );
+  }
+
+  async set(
+    key: string,
+    data: string,
+    options?: { ttl?: number },
+  ): Promise<void> {
+    const { ttl } = Object.assign({}, this.defaultSetOptions, options);
+    await this.client.set(key, data, 'EX', ttl);
```

Comment:
When some cluster nodes down, this client `set` method will thow an exception and [the result](https://github.com/apollographql/apollo-server/blob/master/packages/apollo-datasource-rest/src/HTTPCache.ts#L133) will be lost. So surrounding with try-catch block might be better.

Reply:
I mean. This is a problem that probably already happens, right?
And, in my opinion, should be fixed in [HTTPCache](https://github.com/apollographql/apollo-server/blob/master/packages/apollo-datasource-rest/src/HTTPCache.ts#L125).

Reply:
Right, I think so too.